#140227857 marketplace vendor role

warenga opened the pull request:
#### What does this PR do?
Adds vendor role functionality that allows users to sign up as a vendor.

 #### Description of Task to be completed?

- Users should be able to sign up as a vendor
- Once signed in, a shop should be created
- Vendor's should have the permission to edit their shop settings, create products, edit products, review orders

 #### How should this be manually tested?
Run the deployed version.
Sign up as a vendor.

 #### What are the relevant pivotal tracker stories?
#140227857 Marketplace/ Multi-Vendor: Create a vendor role that enables users to sign up as a vendor

 #### Screenshots (if appropriate)

Vendor registration
<img width="1438" alt="screen shot 2017-02-27 at 4 14 31 pm" src="https://cloud.githubusercontent.com/assets/16874615/23362733/3d3bf256-fd08-11e6-9090-3c0c6d455ac7.png">

Guest user homepage
<img width="1440" alt="screen shot 2017-02-27 at 4 18 13 pm" src="https://cloud.githubusercontent.com/assets/16874615/23362814/8fa82406-fd08-11e6-8ad4-d5320c2c2d27.png">

Vendor user homepage
<img width="1440" alt="screen shot 2017-02-27 at 4 18 44 pm" src="https://cloud.githubusercontent.com/assets/16874615/23362836/b2e07a72-fd08-11e6-9de1-21faf947cd31.png">

Commit:
(feature): add sign up as a vendor option for users

Start 140227857

Commit:
(feature): add role variable in users profile

The variable holds the users account detail i.e whether a normal user or a vendor.

140227857

Commit:
(feature): add meteor method 'user/addUserRole'

This method sets users profile.account to a role.

140227857

Commit:
(feature): change Meteor.call method and argument name

140227857

Commit:
(feature): change Meteor.method name

140227857

Commit:
(feature): assign vendorRoles to vendor users

140227857

Commit:
(feature): add "shop/createVendorShop" method

called when a vendor signs up and creates a shop

140227857

Commit:
(fix): remove unncessary function

140227857

Commit:
(feature): query exisitngshop based on vendorId

140227857

Commit:
(fix): remove console.log()

140227857

Commit:
(feature): add vendorId to shop schema

140227857

Commit:
(fix): change role option from 'Shopper' to 'Customer'

140227857

Commit:
(fix): change role option from 'Shopper' to 'Customer'

140227857

Commit:
(fix): registered currentUser helper to loginDropdown template

140227857

Commit:
(feature): changed customer/vendor dropdown to a radio button

140227857

Commit:
(feature): change customer/vendor role dropdown to a radio button

140227857

Commit:
Merge branch 'develop' into feature/140227857/marketplace-vendor-role

Commit:
(chore): document changes made

140227857

Commit:
(chore): document changes made

140227857

Commit:
(fix): remove unnecessary code

140227857

Commit:
(fix): change meteor.userId() to this.userId()

fix 140227857

Commit:
(fix): remove brackets from this.userId

fix 140227857

Commit:
(fix): add Meteor import

fix 140227857

Commit:
(fix): remover Meteor import

fix 140227857

Commit:
(fix): remove vendor/createShop meteor call

fix 140227857

Commit:
(feature): add shop method that uses factory to create shop

140227857

Commit:
(fix): fix grammar error

140227857

gngchrs commented on client/modules/accounts/templates/signUp/signUp.js:
```diff
@@ -75,6 +90,12 @@ Template.loginFormSignUpView.events({
+  
+        if (role === "vendor") {
+        // creates a new shop when a vendor user signs up
+    
```
What is this 🤷‍♂️

gngchrs commented on server/methods/accounts/accounts.js:
```diff
@@ -442,5 +441,23 @@ Meteor.methods({
+    Meteor.users.update(accountDetails.userId, {
+      $set: { 'profile.account': accountDetails.role }
+    });
+    console.log(accountDetails.role);
```
🙅‍♂️Remove logs while deploying

gngchrs commented on server/methods/core/shop.js:
```diff
@@ -107,6 +158,7 @@ Meteor.methods({
 
     // get currency rates
     result.currency = {};
+    console.log(shop)
```
🙅‍♂️

Commit:
(feature): add shop/createVendorShop meteor call

140227857

Commit:
(fix): remove console.log

140227857

Commit:
(fix): refactor shop/createVendorShop method

140227857

Commit:
(fix) change shop name placeholder

140227857

Commit:
(fix): remove 'shop/createVendorShop' method

It's dependent on next marketplace feature story tasks for it to work properly.

fix 140227857

Commit:
(fix): revert changes made in last commit to fix heroku error

fix 140227857

Commit:
(fix): remove 'shop/createVendorShop' method

fix 140227857

jkanyua commented on client/modules/accounts/templates/signUp/signUp.js:
```diff
@@ -61,13 +62,27 @@ Template.loginFormSignUpView.events({
+      if (err) {
+        alert(err);
+      } else {
+        // success!
```
Why do we have the else case here?

warenga replied:
The else block currently acts as a placeholder for a future story since this is a feature still in development.

jkanyua commented on client/modules/accounts/templates/signUp/signUp.js:
```diff
@@ -61,13 +62,27 @@ Template.loginFormSignUpView.events({
     };
 
-    Accounts.createUser(newUserData, function (error) {
+    Accounts.createUser(newUserData, function(error) {
```
"arrowFunctions": true,

Please lint and see if linter picks up the .eslintrc settings

warenga replied:
Good observation, I will change to arrowFunction and push that change to the next story PR as I've already made a few changes on this file that are related to the next story.

jkanyua commented on server/methods/core/shop.js:
```diff
@@ -57,6 +58,46 @@ Meteor.methods({
+    const shop = createShops();
+
+    shop._id = Random.id();
+    shop.name = 'Shop Name';
```
shop.name = 'Shop Name'

If we are creating a shop, shouldn't the name come from the shop admin? or?

warenga replied:
That is just a placeholder set when a vendor signs up as they are not able to set the shop name during sign up. Though sounds like something that we can add 🤔 . At the moment they should be able to change the name in Shop Settings.

jkanyua reviewed (approved):
Check comments
